sketcherMinimizerConstraintInteraction.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *  sketcherMinimizerConstraintInteraction.h
+ *
+ *  Created by Nicola Zonta on 7/02/2019.
+ *   Copyright Schrodinger, LLC. All rights reserved.
+ *
+ */
+
+#ifndef sketcherMINIMIZERCONSTRAINTINTERACTION
+#define sketcherMINIMIZERCONSTRAINTINTERACTION
+
+#include "sketcherMinimizerInteraction.h"
+
+static const float CONSTRAINT_SCALE = .5f;
+/*force field bond stretches*/
+class sketcherMinimizerConstraintInteraction : public sketcherMinimizerInteraction
+{
+  public:
+    sketcherMinimizerConstraintInteraction(sketcherMinimizerAtom* at1,
+                                        sketcherMinimizerPointF
+                                           position)
+    : sketcherMinimizerInteraction(at1, at1), origin(position) {k = CONSTRAINT_SCALE;};
+    virtual ~sketcherMinimizerConstraintInteraction(){};
+
+    /*calculate the energy of the interaction*/
+    virtual void energy(float& e)
```

Comment:
Does this need to be virtual? I thought that was only necessary on the base. This would be stronger, though, with the override specifier:

    void energy(float& e) override {

Then there would be a compiler error if the override was not successful (e.g., after a change in base class function signature).

Reply:
oh that's nice! thanks